yarn-project/sequencer-client/src/sequencer/abstract_phase_manager.ts

```diff
@@ -243,6 +208,17 @@ export abstract class AbstractPhaseManager {
 
         [kernelOutput, kernelProof] = await this.runKernelCircuit(callData, kernelOutput, kernelProof);
 
+        if (kernelOutput.reverted && this.phase === PublicKernelPhase.APP_LOGIC) {
+          this.log.debug(
+            `Reverting on ${result.execution.contractAddress.toString()}:${functionSelector} with reason: ${
+              result.revertReason
+            }`,
+          );
+          // halt immediately if the public kernel circuit has reverted.
+          // return no logs, as they should not go on-chain.
+          return [kernelOutput, kernelProof, [], result.revertReason];
+        }
```

Comment:
It's possible I missed, but I'm just wondering, should there be an `else if (kernelOutput.reverted) throw new Error` to handle reverts in the non-revertible phases?

Reply:
Yes! I created https://github.com/AztecProtocol/aztec-packages/issues/5038 to track this.